src/CodeGen_D3D12Compute_Dev.cpp

```diff
@@ -0,0 +1,1154 @@
+#include <sstream>
+#include <algorithm>
+#include <iomanip>
+
+#include "CodeGen_D3D12Compute_Dev.h"
+#include "CodeGen_Internal.h"
+#include "Debug.h"
+#include "IROperator.h"
+
+#define DEBUG_TYPES (1)
+
+namespace Halide {
+namespace Internal {
+
+using std::ostringstream;
+using std::string;
+using std::vector;
+using std::sort;
+
+static ostringstream nil;
+
+CodeGen_D3D12Compute_Dev::CodeGen_D3D12Compute_Dev(Target t) :
+    d3d12compute_c(src_stream, t) {
+}
+
+string CodeGen_D3D12Compute_Dev::CodeGen_D3D12Compute_C::print_type_maybe_storage(Type type, bool storage, AppendSpaceIfNeeded space) {
+    ostringstream oss;
+
+    // Storage uses packed vector types.
+    if (storage && type.lanes() != 1)
+    {
+        // TODO(marcos): HLSL has 'packoffset', which only applies to constant buffer fields (constant registers (c))
```

Comment:
Is this something you want to implement before merging?

Reply:
Actually, there's no need to complicate things; in the runtime, prior to the kernel dispatch, constant buffer values are always promoted to a 32bit equivalent. I will update the comment to give more context.